oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java

```diff
@@ -249,14 +256,14 @@ public OSProcess getProcess(int pid) {
             break;
         }
         proc.setProcessID(pid);
-        proc.setParentProcessID(ParseUtil.parseIntOrDefault(split[3], 0));
-        proc.setThreadCount(ParseUtil.parseIntOrDefault(split[19], 0));
-        proc.setPriority(ParseUtil.parseIntOrDefault(split[17], 0));
-        proc.setVirtualSize(ParseUtil.parseLongOrDefault(split[22], 0L));
-        proc.setResidentSetSize(ParseUtil.parseLongOrDefault(split[23], 0L) * this.memoryPageSize);
-        proc.setKernelTime(ParseUtil.parseLongOrDefault(split[14], 0L) * 1000L / hz);
-        proc.setUserTime(ParseUtil.parseLongOrDefault(split[13], 0L) * 1000L / hz);
-        proc.setStartTime(bootTime + ParseUtil.parseLongOrDefault(split[21], 0L) * 1000L / hz);
+        proc.setParentProcessID(ParseUtil.parseIntOrDefault(split[3 + nameOffset], 0));
+        proc.setThreadCount(ParseUtil.parseIntOrDefault(split[19 + nameOffset], 0));
+        proc.setPriority(ParseUtil.parseIntOrDefault(split[17 + nameOffset], 0));
+        proc.setVirtualSize(ParseUtil.parseLongOrDefault(split[22 + nameOffset], 0L));
+        proc.setResidentSetSize(ParseUtil.parseLongOrDefault(split[23 + nameOffset], 0L) * this.memoryPageSize);
+        proc.setKernelTime(ParseUtil.parseLongOrDefault(split[14 + nameOffset], 0L) * 1000L / hz);
+        proc.setUserTime(ParseUtil.parseLongOrDefault(split[13 + nameOffset], 0L) * 1000L / hz);
+        proc.setStartTime(bootTime + ParseUtil.parseLongOrDefault(split[21 + nameOffset], 0L) * 1000L / hz);

```

Comment:
similar to above comment, perhaps ensure `split` length exceeds `21 + nameOffset`

Reply:
These are god points.
I'll add some better array length and string content checking, plus run the code through a code formatter.

Reply:
Other thing I've been thinking is that it may not be necessary to check for the ") " on the name.  We know there will be 52 fields, so if the split length is >52 we can subtract 52 to know how big the offset is (and how many fields to append to the name.)   This could wrap the length check in with a useful determination of nameOffset.  (And then you could piece together the name and strip off first and last characters in one go.)